### PR TITLE
feat: refine claim offsets

### LINKED ISSUE #101: Offsets are based on chunks, which are really large
As @dearden points out in #95, we can use the direct quote of the claim together with the original transcript to refine the start and end offsets.

### PR DESCRIPTION
**Pull request by andylolz**
Fixes #101.

-----------------

## Pull request checklist

- [x] I have linked my PR to an issue
- [x] I’ve used [conventional commits](https://github.com/FullFact/automation-docs/wiki/Conventional-commits)
- [x] My branch is up-to-date with `main`
- [ ] Where appropriate, I have added or updated tests
- [ ] Where appropriate, I have [updated documentation](https://github.com/FullFact/automation-docs/) to reflect my changes

### CODE REVIEW

**Review by ff-dh — APPROVED** [review]

File: src/raphael_backend_flask/process.py [hunk 1]
```diff
@@ -30,6 +30,43 @@ def download_transcript(youtube_id: str) -> int:
...
+    # refine the end of the chunk
+    chunk_text = ""
+    for end_idx, sentence in enumerate(transcript[start_idx:end_idx], start_idx + 1):
+        chunk_text += sentence['sentence_text'] + " "
+        if claim["raw_sentence_text"] in chunk_text:
+            break
+    else:
+        # Error! Couldn’t find raw sentence in transcript.
+        # This means the LLM is not returning the raw sentence text.
+        return claim
```
Comment (andylolz) [inline src/raphael_backend_flask/process.py:54]: Just to note here… I don’t know if we should be logging something here… Currently when this happens, we just ignore it. Maybe that is fine – I know @dearden is interested in looking at this in experiments, so maybe he’s fine with us ignoring this here.

**Comment by andylolz** [comment]
Many thanks for the review!

I’ll merge, but with the caveat that I’m conscious this in particular is **screaming out** for some tests. I suspected there could be edge case problems with this. I’m reasonably happy that there aren’t, but I can’t be 100% sure without tests.
